src/ember-app.js

```diff
@@ -365,22 +319,27 @@ class EmberApp {
       }, destroyAppInstanceInMs);
     }
 
-    return this.visitRoute(path, fastbootInfo, bootOptions, result)
-      .then(() => {
-        if (!disableShoebox) {
-          // if shoebox is not disabled, then create the shoebox and send API data
-          createShoebox(doc, fastbootInfo);
-        }
-      })
-      .catch(error => (result.error = error))
-      .then(() => result._finalize())
-      .finally(() => {
-        if (result._destroyAppInstance()) {
-          if (destroyAppInstanceTimer) {
-            clearTimeout(destroyAppInstanceTimer);
-          }
+    try {
+      await this.visitRoute(path, fastbootInfo, bootOptions, result);
+
+      if (!disableShoebox) {
+        // if shoebox is not disabled, then create the shoebox and send API data
+        createShoebox(doc, fastbootInfo);
+      }
+
+      result._finalize();
+    } catch (error) {
+      // eslint-disable-next-line require-atomic-updates
+      result.error = error;
+    } finally {
+      if (result._destroyAppInstance()) {
+        if (destroyAppInstanceTimer) {
+          clearTimeout(destroyAppInstanceTimer);
         }
-      });
+      }
+    }
```

Comment:
Given we are creating a new `EmberApp` every `visit` now, this means that we can only have 1 App Instance per `EmberApp` rather previously n EmberAppInstances per EmberApp. Please correct me if I am misunderstanding this. 

Given the above, we should also destroy the `EmberApp` after every `visit` (by calling and fixing the `destroy` method) so that any timer related code doesn't run until the next `visit` request.  Otherwise we will end up running async code without a live app instance and not be able to attribute those async errors that occur after visit correctly.

Reply:
> Given the above, we should also destroy the EmberApp and the sandbox after every visit (by calling and fixing the destroy method) so that any timer related code doesn't run until the next visit request.

Yep, totally correct. Will update to ensure we call destroy.

Reply:
> Given we are creating a new EmberApp every visit now, this means that we can only have 1 App Instance per EmberApp at any given point of time rather previously n EmberAppInstances per EmberApp. 

Actually this isn't _quite_ correct, but the terminology actually confused me. 😄 I'm going to refer to "ember app" as the actual app that is booted (from `distPath`) and "fastboot ember app" as this local file (`src/ember-app.js`). 

The "fastboot ember app" is instantiated once (by the main entry point in `src/index.js`) when the end user calls `let fastboot = new FastBoot({ distPath })` and is destroyed then re-created whenever the user calls `fastboot.reload()`. The "ember app" will be instantiated once per `visit` call, and destroyed upon completion.

Reply:
OK, just pushed a commit that ensures both `Ember.Application` and `Ember.ApplicationInstance` are destroyed after the visit concludes.

Reply:
OK, just pushed a commit that ensures both Ember.Application and Ember.ApplicationInstance are destroyed after the visit concludes.